lib/fs.ts

```diff
@@ -26,16 +26,21 @@ import * as pathModule from 'path';
 import {ReadStream, WriteStream} from './streams';
 
 const ROOT_PATH = pathModule.resolve(__dirname, '..');
+const DEFAULT_THROTTLE = 5 * 1000; // 5 seconds
 
-interface PendingUpdate {
+export interface PendingUpdate {
 	isWriting: boolean; // true: waiting on a call to FS.write, false: waiting on a throttle
 	pendingDataFetcher: (() => string | Buffer) | null;
 	pendingOptions: AnyObject | null;
 	throttleTime: number; // throttling until time (0 for no throttle)
 	throttleTimer: NodeJS.Timer | null;
 }
 
-const pendingUpdates = new Map<string, PendingUpdate>();
+if (!global.__fsState) {
+	global.__fsState = {
+		pendingUpdates: new Map<string, PendingUpdate>(),
+	};
```

Comment:
```suggestion
declare const global: {__fsState: {pendingUpdates: Map<string, PendingUpdate>}};

if (!global.__fsState) {
	global.__fsState = {
		pendingUpdates: new Map(),
	};
```

This _should_ work to locally override `global`.